pkg/parse/parse.go

```diff
@@ -334,304 +334,51 @@ func getMounts(mounts []string, contextDir string) (map[string]specs.Mount, erro
 	for _, mount := range mounts {
 		arr := strings.SplitN(mount, ",", 2)
 		if len(arr) < 2 {
-			return nil, errors.Wrapf(errInvalidSyntax, "%q", mount)
+			return nil, mountedImages, errors.Wrapf(errInvalidSyntax, "%q", mount)
 		}
 		kv := strings.Split(arr[0], "=")
 		// TODO: type is not explicitly required in Docker.
 		// If not specified, it defaults to "volume".
 		if len(kv) != 2 || kv[0] != "type" {
-			return nil, errors.Wrapf(errInvalidSyntax, "%q", mount)
+			return nil, mountedImages, errors.Wrapf(errInvalidSyntax, "%q", mount)
 		}
 
 		tokens := strings.Split(arr[1], ",")
 		switch kv[1] {
 		case TypeBind:
-			mount, err := GetBindMount(tokens, contextDir)
+			mount, image, err := internalParse.GetBindMount(nil, tokens, contextDir, store, "", nil)
```

Comment:
This still needs to not be passing `nil` in as a `SystemContext`.

Reply:
`getMounts()` and `GetVolumes()` already have changes being made to their signatures to allow passing in a `Store` object, so in terms of API compatibility we've already broken anyone who used `GetVolumes()`.  Have their callers pass in a `SystemContext`, too.

Reply:
@nalind Following function is being invoked from `cmd/buildah/run.go` when we do `build run --mount` we don't have any `github.com/containers/image/v5/types".SystemContext` there. Could i pass empty context from there ?